lib/src/commit_builder.rs

```diff
@@ -33,6 +34,7 @@ impl CommitBuilder<'_> {
         settings: &UserSettings,
         parents: Vec<CommitId>,
         tree_id: TreeId,
+        tree_conflict_format: bool,
```

Comment:
Optional: It might be slightly better to do one of the following:

1. remove the `tree_conflict_format` argument and always make it `false`, since we use `from_legacy_tree_id`. A different function could create the ones with tree_conflict_format = true
2. Or keep the `tree_conflict_format` argument but have the function take `Conflict<TreeId>`.